dask_ml/model_selection/_successive_halving.py

```diff
@@ -0,0 +1,220 @@
+import math
+
+import numpy as np
+import toolz
+
+from ._incremental import IncrementalSearchCV
+
+
+class SuccessiveHalvingSearchCV(IncrementalSearchCV):
+    """
+    Perform the successive halving algorithm [1]_.
+
+    This algorithm trains estimators for a certain number ``partial_fit``
+    calls to ``partial_fit``, then kills the worst performing half.
+    It trains the surviving estimators for twice as long, and repeats this
+    until one estimator survives.
+
+    The value of :math:`1/2` above is used for a clear explanation. This class
+    defaults to killing the worst performing ``1 - 1 // aggressiveness``
+    fraction of models, and trains estimators for ``aggressiveness`` times
+    longer, and waits until the number of models left is less than
+    ``aggressiveness``.
+
+    Parameters
+    ----------
+    estimator : estimator object.
+        A object of that type is instantiated for each initial hyperparameter
+        combination. This is assumed to implement the scikit-learn estimator
+        interface. Either estimator needs to provide a ``score`` function,
+        or ``scoring`` must be passed. The estimator must implement
+        ``partial_fit``, ``set_params``, and work well with ``clone``.
+
+    parameters : dict
+        Dictionary with parameters names (string) as keys and distributions
+        or lists of parameters to try. Distributions must provide a ``rvs``
+        method for sampling (such as those from scipy.stats.distributions).
+        If a list is given, it is sampled uniformly.
+
+    aggressiveness : float, default=3
+        How aggressive to be in culling off the different estimators. Higher
+        values imply higher confidence in scoring (or that
+        the hyperparameters influence the ``estimator.score`` more
+        than the data).
+
+    n_initial_parameters : int, default=10
+        Number of parameter settings that are sampled.
+        This trades off runtime vs quality of the solution.
+
+    n_initial_iter : int
+        Number of times to call partial fit initially before scoring.
+        Estimators are trained
+        for ``n_initial_iter`` calls to ``partial_fit`` at first.
+
+    max_iter : int, default 100
+        Maximum number of partial fit calls per model.
+
+    test_size : float
```

Comment:
If `test_size` is used, these docstrings should use "test" instead of "hold out" or "validation"